### ISSUE TITLE
Question: MSSql models

### ISSUE THREAD

**Issue by ArieJones** [eventflow/EventFlow#250]
I was wondering if there were any examples and/or guidance on how to handle the migrations when using SQL Server? 

Specifically, does it hurt calling MigrateDatabase on startup just by default?
 I am assuming that the framework would know whether or not it had something that needed updated... 

Thanks,
AJ

**Comment by rasmus** [comment]
@ArieJones Deploy SQL migrations as a separate step during service install for our micro services. We use Octopus Deploy for deployment and the migrations are simply a step. It should be possible to apply EventFlow SQL migrations while the site is active and running the previous versions.

However, EventFlow only has a two SQL scripts for the initial MSSQL event store setup. I don't expect to add any anytime soon, unless there's a bug.

If you're using MSSQL as read store it gets a bit more complicated. I would recommend doing updates to read models that are "additive", i.e., new models are backward compatible. Then you can apply the database changes while running the system and then update the application after. If you need to do breaking changes, create a new model.

Basically this isn't any different from any other MSSQL schema change scenario.

Does that answer your question?

**Comment by ArieJones** [comment]
@Ramus 
Yes thanks for that answer... very intrigued by your hard work here.
